File: backend/Margorak.Api/Dto/CharacterDto.cs
using Margorak.Api.Enums;

namespace Margorak.Api.Dto
{
    public class CharacterDto
    {
        public int Id { get; set; } 
        public string Name { get; set; } = string.Empty;
        public Gender Gender { get; set; }
        public CharacterRaceDto CharacterRace { get; set; } = null!;
        public CharacterClassDto CharacterClass { get; set; } = null!;
        public List<OwnedItemDto> OwnedItems { get; set; } = [];
        public List<CharacterEquipmentDto> CharacterEquipment {  get; set; } = [];
        public int Level {  get; set; }
        public int Experience {  get; set; }
        public int StatusPoints { get; set; }
        public int Strength { get; set; }
        public int Dexterity { get; set; }
        public int Vitality { get; set; }
        public int Intelligence { get; set; }
        public int CurrentHp { get; set; }
        public int MaxHp { get; set; }
        public int CurrentMp { get; set; }
        public int MaxMp { get; set; }
        public int Gold { get; set; }
        public int CurrentMapId { get; set; }
        public int LocX { get; set; }
        public int LocY { get; set; }
        public int Version { get; set; }
    }
}

File: frontend/src/app/features/character/dto/character.dto.ts
export interface CharacterDto {
  id: number;
  name: string;
  gender: string;

  characterRace: CharacterRaceDto;
  characterClass: CharacterClassDto;

  level: number;
  experience: number;
  statusPoints: number;

  strength: number;
  dexterity: number;
  vitality: number;
  intelligence: number;

  currentHp: number;
  maxHp: number;
  currentMp: number;
  maxMp: number;

  gold: number;

  currentMapId: number;
  locX: number;
  locY: number;

  version: number;
}
export interface CharacterRaceDto {
  id: number;
  name: string;
  strengthMod: number;
  dexterityMod: number;
  vitalityMod: number;
  intelligenceMod: number;
}

export interface CharacterClassDto {
  id: number;
  name: string;
}

export interface CharacterStatsDto {
  strength: number;
  dexterity: number;
  intelligence: number;
  vitality: number;
  statusPoints: number;
}

export type CharacterStat = 'dexterity' | 'strength' | 'intelligence' | 'vitality';
